src/pages/main-page/components/products-section/components/books-list/BooksList.tsx

```diff
@@ -0,0 +1,81 @@
+import { useEffect, useState } from 'react';
+import styles from './BooksList.module.scss';
+import { messages } from '@/sources/messages';
+import { BookCard } from '../../../../../../components/books-cards/BookCard';
+import type { BookData } from '@/sources/types';
+import { fetchBooksData } from '@/api/fetchBooksData';
+import { Pagination } from '@/components/pagination/Pagination';
+import { useSearchParams } from 'react-router-dom';
+
+export interface Props {
+  searchTerm: string;
+  setLoading: (value: boolean) => void;
+  onClose: () => void;
+  isLoading: boolean;
+  setError: (message: string) => void;
+}
+
+export const BooksList = ({ searchTerm, setLoading, setError }: Props) => {
+  const [books, setBooks] = useState<BookData[]>([]);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [totalItems, setTotalItems] = useState(0);
+
+  const pageItemsResults = 5;
+  const totalPages = Math.ceil(totalItems / pageItemsResults);
+  const pageFromURL = Number.parseInt(searchParams.get('page') || '1', 10);
```

Comment:
```suggestion
  const pageFromURL = Number(searchParams.get('page') || '1') || 1;
```
ваше не критично, просто как будто понятнее. ну или нет)))